apps/omg/lib/omg/state/transaction/fee.ex

```diff
@@ -0,0 +1,114 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.State.Transaction.Fee do
+  @moduledoc """
+  Internal representation of a fee claiming transaction in plasma chain.
+  """
+  alias OMG.Crypto
+  alias OMG.Output
+  alias OMG.State.Transaction
+
+  @fee_output_type_marker <<0xFE, 0xE0>>
+  @uniqueness_output_type_marker <<0xF0, 0x0F>>
+
+  defstruct [:outputs]
+
+  @type t() :: %__MODULE__{outputs: [Output.FungibleMoreVPToken.t() | Output.UniquenessEnforcer.t()]}
+
+  @doc """
+  Creates new fee claiming transaction
+  """
+  @spec new(
+          blknum :: non_neg_integer(),
+          {Crypto.address_t(), Transaction.Payment.currency(), pos_integer}
+        ) :: t()
+  def new(blknum, {owner, currency, amount}) do
+    %__MODULE__{
+      outputs: [
+        %Output.UniquenessEnforcer{blknum: blknum, type_marker: @uniqueness_output_type_marker},
+        %Output.FungibleMoreVPToken{
+          owner: owner,
+          currency: currency,
+          amount: amount,
+          type_marker: @fee_output_type_marker
+        }
+      ]
+    }
+  end
+
+  @doc """
+  Transaform the structure of RLP items after a successful RLP decode of a raw transaction, into a structure instance
+  """
+  def reconstruct([outputs_rlp]) do
+    with {:ok, outputs} <- reconstruct_outputs(outputs_rlp),
+         do: {:ok, %__MODULE__{outputs: outputs}}
+  end
+
+  def reconstruct(_), do: {:error, :malformed_transaction}
+
+  defp reconstruct_outputs(outputs_rlp) do
+    with {:ok, outputs} <- parse_outputs(outputs_rlp),
```

Comment:
I know this got ported from Payment, we can cleanup payment separately, but this `with` is pointless, as is the distinction between `parse` and `reconstruct`. We can simplify here